src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -125,7 +135,13 @@ citus_table_size(PG_FUNCTION_ARGS)
 		tableSizeFunction = CSTORE_TABLE_SIZE_FUNCTION;
 	}
 
-	uint64 tableSize = DistributedTableSize(relationId, tableSizeFunction);
+	uint64 tableSize = 0;
+
+	if (!DistributedTableSize(relationId, tableSizeFunction, failOnError, &tableSize))
+	{
+		/* should always throw error */
+		Assert(false);
```

Comment:
Can we have Assert(!failOnError); here as well? Ideally I would prefer an error here but I guess that is longer/independent discussion.